### DISCUSSION TITLE
how to change layout page ?

### DISCUSSION THREAD

**Discussion by magic-thomas** [filamentphp/filament#8989] [Help]
Hi. 
Default layout is 'resources/views/components/layouts/app.blade.php' 

How can I change this ? 

I want to use 'resources/views/components/layouts/app-filament.blade.php' for filament.

**Comment by leandrocfe** [comment]
Would you like to create a custom page with Filament layout? Why not a [custom page](https://filamentphp.com/docs/3.x/panels/pages)?

**Comment by eugenefvdm** [comment]
I'm trying to create a custom page. I'm trying to use Flux.

Flux requires this:

```
<head>
    ...
    @fluxStyles
</head>

<body>
    ...
    @fluxScripts
</body>
```

So I need to update the "default master layout", or whatever it's called.

By all accounts, this has something to do with `base.blade.php`

When you publish Filament 3 assets, it doesn't publish the `layouts` directory.

Next, I've tried to copy it over from the `vendor` folder.

I tried copying `base.blade.php` to:

`resources/views/vendor/filament/layouts`
`resources/views/vendor/filament/layout`
`resources/views/layouts`

I'm at wits end. Please assist.

**Reply by abduselam1** [reply]
Hey @eugenefvdm did you manage to update the layout?

**Reply by leandrocfe** [reply]
You can use Filament render hooks

AppServiceProvider.php
```php
public function boot(): void
{
    FilamentView::registerRenderHook(
        PanelsRenderHook::STYLES_BEFORE,
        fn (): string => Blade::render('@vite(\'resources/css/app.css\')'),
    );

    FilamentView::registerRenderHook(
        PanelsRenderHook::STYLES_BEFORE,
        fn (): string => Blade::render('@fluxStyles'),
    );

    FilamentView::registerRenderHook(
        PanelsRenderHook::BODY_END,
        fn (): string => Blade::render('@fluxScripts'),
    );
}
```

**Reply by rasul-chopurov** [reply]
> You can use Filament render hooks
> 
> AppServiceProvider.php
> 
> ```
> public function boot(): void
> {
>     FilamentView::registerRenderHook(
>         PanelsRenderHook::STYLES_BEFORE,
>         fn (): string => Blade::render('@vite(\'resources/css/app.css\')'),
>     );
> 
>     FilamentView::registerRenderHook(
>         PanelsRenderHook::STYLES_BEFORE,
>         fn (): string => Blade::render('@fluxStyles'),
>     );
> 
>     FilamentView::registerRenderHook(
>         PanelsRenderHook::BODY_END,
>         fn (): string => Blade::render('@fluxScripts'),
>     );
> }
> ```

now flux used tailwind css 4.0 after install this
filament php not work how to fix ?
not help me 

        FilamentAsset::register([
            Js::make('tailwindcss', 'https://cdn.tailwindcss.com/3.4.16'),
        ]);
